src/casl/casl-ability.factory.ts

```diff
@@ -783,96 +784,198 @@ export class CaslAbilityFactory {
        */
 
       // -------------------------------------
-      // jobs
+      // loop through the jobs definition and extract the most permissive permission
       // -------------------------------------
       // endpoint authorization
-      cannot(AuthOp.JobsRead, JobClass);
-      cannot(AuthOp.JobsCreate, JobClass);
-      cannot(AuthOp.JobsUpdate, JobClass);
-    } else if (
-      user.currentGroups.some((g) => configuration().adminGroups.includes(g))
-    ) {
-      /**
-       * authenticated users belonging to any of the group listed in ADMIN_GROUPS
-       */
-
-      // -------------------------------------
-      // jobs
-      // -------------------------------------
-      // endpoint authorization
-      can(AuthOp.JobsRead, JobClass);
-      can(AuthOp.JobsCreate, JobClass);
-      can(AuthOp.JobsUpdate, JobClass);
-      // -------------------------------------
-      // data instance authorization
-      can(AuthOp.JobsReadAny, JobClass);
-      can(AuthOp.JobsCreateAny, JobClass);
-      can(AuthOp.JobsUpdateAny, JobClass);
-    } else if (
-      user.currentGroups.some((g) =>
-        configuration().createJobGroups.includes(g),
-      )
-    ) {
-      /**
-       * authenticated users belonging to any of the group listed in CREATE_JOBS_GROUPS
-       */
+      // job creation
+      if (
+        configuration().jobConfiguration.some(
+          (j) => j.create.auth == CreateJobAuth.All,
+        )
+      ) {
+        can(AuthOp.JobCreate, JobClass);
+      } else {
+        cannot(AuthOp.JobCreate, JobClass);
+      }
+      cannot(AuthOp.JobRead, JobClass);
+      if (
+        configuration().jobConfiguration.some(
+          (j) => j.update.auth == UpdateJobAuth.All,
+        )
+      ) {
+        can(AuthOp.JobStatusUpdate, JobClass);
+      } else {
+        cannot(AuthOp.JobStatusUpdate, JobClass);
+      }
+      cannot(AuthOp.JobDelete, JobClass);
 
       // -------------------------------------
-      // jobs
-      // -------------------------------------
-      // endpoint authorization
-      can(AuthOp.JobsRead, JobClass);
-      can(AuthOp.JobsCreate, JobClass);
-      can(AuthOp.JobsUpdate, JobClass);
-      // -------------------------------------
-      // data instance authorization
-      can(AuthOp.JobsCreateAny, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
+      // instance authorization
+      can(AuthOp.JobCreateConfiguration, JobClass, {
+        "configuration.create.auth": CreateJobAuth.All,
       });
-      can(AuthOp.JobsReadAccess, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
+      can(AuthOp.JobCreateConfiguration, JobClass, {
+        "configuration.create.auth": CreateJobAuth.DatasetPublic,
```

Comment:
I don't think this syntax is not the best approach. Same as the syntax using `?` that I mentioned in another comment, I believe it leaves room for ambiguity and potential errors

Reply:
@despadam can you make a suggestion on how to improve it?

Reply:
Let me think about it and I will try to come up with something. However, in the meantime, this syntax is responsible for some of the compilation errors that we get. Can we somehow patch it for now?